### DISCUSSION TITLE
Some functional suggestions I have summarized, I hope they can be adopted

### DISCUSSION THREAD

**Discussion by 2dogtwo** [godotengine/godot-proposals#4884] [General Discussions]
### Describe the project you are working on

A tool software

### Describe the problem or limitation you are having in your project

1. The self-contained sub-nodes can be converted into custom nodes. You can modify them according to your needs, just like UNITY. Although godot looks very simple now, it is really troublesome to modify the nodes. . My word is that you can only remake a node with similar functions yourself.
2. The theme hopes to add the function of identifying child nodes. This is actually for the child nodes that come with the editor. For example, the theme of OptionButton needs to modify PopupMenu, which is very confusing.
3. The movement range of the sub-window cannot exceed the scope of the main window. This is a big problem. I thought I would not use it at first, until I found that the editor itself is like this, and the pop-up window should not be able to move at will, right? If the button of the child window is accidentally covered. You can't close the child window by other methods, otherwise the program is equivalent to being stuck.
of.
4. The editor supports mouse drag and drop operation. A node is declared in the script. Selection is only possible through the corresponding window in the editor. But for someone like me who doesn't like naming nodes specifically, this window selection always leaves me a little confused about which is which. It would be nice if you could just drag and drop.
5. Error message, hope to be clear. Some error messages make it impossible to see where the problem is. Especially the error message about the engine itself.
6. The help file can search the content, this is for the help file in the engine.

So much for now. That's all I can remember. The above content is from Google Translate, I hope you can understand it.

**Comment by 2dogtwo** [comment]
now i use version is 3.4

**Comment by Calinou** [comment]
> The movement range of the sub-window cannot exceed the scope of the main window. This is a big problem. I thought I would not use it at first, until I found that the editor itself is like this, and the pop-up window should not be able to move at will, right? If the button of the child window is accidentally covered. You can't close the child window by other methods, otherwise the program is equivalent to being stuck.

This is resolved in 4.0.alpha thanks to multi-window support. It's enabled by default in the editor, but disabled by default in projects as most games don't need to spawn more than one window. To enable multi-window support in project, disable **Embed Subwindows** in the Project Settings.

Multi-window relies on a backwards-incompatible DisplayServer refactoring, so it won't be backported to 3.x.

> Error message, hope to be clear. Some error messages make it impossible to see where the problem is. Especially the error message about the engine itself.

Can you list some examples of those error messages in https://github.com/godotengine/godot/issues/42719?

> The help file can search the content, this is for the help file in the engine.

Pressing <kbd>Ctrl + F</kbd> while on an editor help page should allow you to search in the editor help already.

**Reply by 2dogtwo** [reply]
All right. thanks for the reply. I know that 4.0 can have multiple windows, so I replied that the version I used was 3.4, so what about the rest. Can it be realized?
